graphstorm-processing/graphstorm_processing/graph_loaders/dist_heterogeneous_loader.py

```diff
@@ -622,6 +627,67 @@ def csv_row(data: Row):
         )
         return filtered_key_list
 
+    @staticmethod
+    def _create_metadata_entry(path_list: Sequence[str]) -> dict:
+        return {
+            "format": {"name": FORMAT_NAME, "delimiter": DELIMITER},
+            "data": path_list,
+        }
+
+    # TODO: Enable writing table to multiple Parquet files
+    def _write_pyarrow_table(self, pa_table: pa.Table, out_path: str) -> list[str]:
+        """Writes a single PyArrow Table to Parquet.
+
+        Parameters
+        ----------
+        pa_table : pa.Table
+            PyArrow Table to write to storage.
+        out_path : str
+            Full filepath to write. Can be an S3 URI or local path.
+
+        Returns
+        -------
+        list[str]
+            The list of files written, paths being relative to ``self.output_prefix``.
+        """
+
+        # TODO: Remove once we support writing multiple files
+        assert out_path.endswith(".parquet")
+        if self.filesystem_type == FilesystemType.LOCAL:
+            os.makedirs(os.path.dirname(out_path), exist_ok=True)
+
+        pq.write_table(
```

Comment:
Shall we move this part to line 666? And if emr-serverless support directly use pyarrow to write to S3?

Reply:
We can move it yes. 

> emr-serverless support directly use pyarrow to write to S3

Yes we already use pyarrow to write and modify files on EMRS during the re-partition stage that runs on the Spark leader.